broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.partitioning;
+
+import io.atomix.raft.partition.RaftPartitionGroup;
+import io.atomix.raft.partition.RaftPartitionGroup.Builder;
+import io.camunda.zeebe.broker.clustering.ClusterServices;
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.configuration.ClusterCfg;
+import io.camunda.zeebe.broker.system.configuration.DataCfg;
+import io.camunda.zeebe.broker.system.configuration.NetworkCfg;
+import io.camunda.zeebe.logstreams.impl.log.ZeebeEntryValidator;
+import io.camunda.zeebe.snapshots.ReceivableSnapshotStoreFactory;
+import io.camunda.zeebe.util.FileUtil;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.agrona.IoUtil;
+
+public class PartitionManagerFactory {
+
+  public static final String GROUP_NAME = "raft-partition";
+
+  public static PartitionManager fromBrokerConfiguration(
+      final BrokerCfg configuration,
+      final ClusterServices clusterServices,
+      final ReceivableSnapshotStoreFactory snapshotStoreFactory) {
+
+    final var partitionGroup = buildRaftPartitionGroup(configuration, snapshotStoreFactory);
+
+    return new PartitionManager(
+        partitionGroup,
+        clusterServices.getMembershipService(),
+        clusterServices.getCommunicationService());
+  }
+
+  private static RaftPartitionGroup buildRaftPartitionGroup(
+      final BrokerCfg configuration, final ReceivableSnapshotStoreFactory snapshotStoreFactory) {
+
+    final DataCfg dataConfiguration = configuration.getData();
+    final String rootDirectory = dataConfiguration.getDirectory();
+    try {
+      FileUtil.ensureDirectoryExists(new File(rootDirectory).toPath());
+    } catch (final IOException e) {
+      throw new UncheckedIOException("Failed to create data directory", e);
+    }
+
+    final File raftDirectory = new File(rootDirectory, GROUP_NAME);
```

Comment:
:wrench: Similarly, we can reuse the path constructed above and just do `rootPath.resolve(GROUP_NAME)` to get the new path. I would also propose we replace the `IoUtil.ensureDirectoryExists` usage with `FileUtil.ensureDirectoryExists` as above - we had an issue to replace all their usages, but I guess we missed one. The `IoUtil` variant unfortunately hides the error - it simply says if it failed or not, but you don't know why, which is why we decided recently to always go for the `FileUtil` variant.

I know it's a bit out of scope since you just extracted this code, so I'm happy to have this in a follow up.